src/distilabel/llm/anthropic.py

```diff
@@ -0,0 +1,124 @@
+    api_key: Optional[SecretStr] = os.getenv("ANTHROPIC_API_KEY", None)  # type: ignore
+    base_url: Union[str, None] = None
+    timeout: float = 600.0
+    http_client: Union[str, None] = None
+    max_retries: int = 2
+
+    _env_var: Optional[str] = PrivateAttr(default="ANTHROPIC_API_KEY")
+    _aclient: Optional["AsyncAnthropic"] = PrivateAttr(...)
+
+    def load(self, api_key: Optional[str] = None) -> None:
+        """Loads the `AsyncAnthropic` client to use the Anthropic async API."""
+        try:
+            from anthropic import AsyncAnthropic
+        except ImportError as ie:
+            raise ImportError(
+                "Anthropic Python client is not installed. Please install it using"
+                " `pip install anthropic`."
+            ) from ie
+
+        self.api_key = self._handle_api_key_value(
+            self_value=self.api_key,
+            load_value=api_key,
+            env_var=self._env_var,  # type: ignore
+        )
+
+        self._aclient = AsyncAnthropic(
+            api_key=self.api_key.get_secret_value(),
+            base_url=self.base_url,
+            timeout=self.timeout,
+            http_client=self.http_client,
+            max_retries=self.max_retries,
+        )
+
+    @property
+    def model_name(self) -> str:
+        """Returns the model name used for the LLM."""
+        return self.model
+
+    async def agenerate(  # type: ignore
+        self,
+        input: "ChatType",
+        system: str = "",
+        num_generations: int = 1,
+        max_tokens: int = 128,
+        stop_sequences: List[str] = None,
+        temperature: float = 1.0,
+        top_p: float = 1.0,
+        top_k: int = 0,
+    ) -> "GenerateOutput":
+        """Generates a response asynchronously, using the [Anthropic Async API definition](https://github.com/anthropics/anthropic-sdk-python).
+
+        Args:
+            input: a single input in chat format to generate responses for.
+            system: the system prompt to use for the generation. No existing `system` role within the input conversation, only `user` and `assistant`. Defaults to `""`.
+            num_generations: the number of generations to create per input. Defaults to `1`.
+            max_tokens: the maximum number of new tokens that the model will generate. Defaults to `128`.
+            stop_sequences: custom text sequences that will cause the model to stop generating. Defaults to None.
+            temperature: the temperature to use for the generation. Set only if top_p is None. Defaults to `1.0`.
+            top_p: the top-p value to use for the generation. Defaults to `1.0`.
+            top_k: the top-k value to use for the generation. Defaults to `0`.
+
+        Returns:
+            A list of lists of strings containing the generated responses for each input.
+        """
+
+        completion = await self._aclient.messages.create(
+            model=self.model,
+            system=system,
```

Comment:
try to get the system prompt from the first element of `input` if role == system. Otherwise, leave client default value.